src/utils/moment.js

```diff
@@ -71,18 +75,25 @@ export default async function loadMomentLocalization() {
  */
 async function getLocaleFor(locale) {
 	try {
-		// default load e.g. fr-fr
+		// default load e.g.en-de
 		await import('moment/locale/' + locale)
 		return locale
 	} catch (error) {
 		try {
-			// failure: fallback to fr
-			locale = locale.split('-')[0]
+			// failure: fallback to de
+			locale = locale.split('-')[1]
 			await import('moment/locale/' + locale)
 			return locale
 		} catch (e) {
-			// failure, fallback to english
-			console.debug('Fallback to locale', 'en')
+			try {
+				// failure: fallback to en
+				locale = locale.split('-')[0]
+				await import('moment/locale/' + locale)
+				return locale
+			} catch (e) {
+				// failure, fallback to english
+				console.debug('Fallback to locale', 'en')
+			}
```

Comment:
I don't quite get this change. Could you please elaborate on it?

Why change the order of parsing locales (e.g. en-de: de -> en instead of en -> de)?

Reply:
This is the core of the problem. The locale here is loaded. If it doesn't exist, it will fallback. Right now, with en-de, English will be the locale and thus, the date formatting will be the moment 'en' locale. This is why right now, setting the locale to en-de, the starting day will be on Sunday. Most users see this as a bug.

This is why I think that the ultimate problem is the settings. I will illustrate. If you are German and want the language be English but date formatting in German, you can do all these combinations:

| Language       | Locale     | Expected | Current actual result |
|--------------|-----------|------------| ------ |
| English | German (de)    | German dates, English text  | German dates, English text
| English      | German English (en-de)  | German dates, English text       | English dates, English text

This system means you can also do weird combinations:

| Language       | Locale     | Expected | Current actual result |
|--------------|-----------|------------|------ |
| English | Monocan French  (fr-mc) | ??  | French dates, English text |

As you can see, it currently ignores the locale part of the language-locale string. This is why a temporary solution is to ignore the language part, as that is already encoded as the language setting, but ultimately I think that you should only be able to select countries in the locale menu, unless I am missing something.

Reply:
Ahh, now I understand the issue. Thanks for elaborating :)

I found a tiny bug in your code and pushed a fixup. Please take a look. The already split locale was split another time in the fallback logic and thus the second fallback was never working as intended.

Reply:
I tested the original implementation in JSfiddle and it should also work: this is because it splits the locale parameter multiple times instead of the splitLocale. It is confusing though, so thank you for the refactor